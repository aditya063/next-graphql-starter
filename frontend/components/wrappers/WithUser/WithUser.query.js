import gql from 'graphql-tag';

export const ME_QUERY = gql`
  query ME_QUERY {
    me {
      id
      email
    }
  }
`;

export const MOCK_ME_QUERY = {
  request: {
    query: ME_QUERY
  },
  result: {
    data: {
      me: {
        id: '1',
        email: 'mock@email.com'
      }
    }
  }
};
