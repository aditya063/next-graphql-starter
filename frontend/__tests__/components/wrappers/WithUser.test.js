import React from 'react';
import { shallow } from 'enzyme';
import WithUser from '../../../components/wrappers/WithUser/WithUser';

describe('WithUser', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<WithUser {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
