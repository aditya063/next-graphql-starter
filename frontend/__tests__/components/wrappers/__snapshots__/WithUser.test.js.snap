// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithUser matches snapshot 1`] = `
<Query
  errorPolicy="all"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "ME_QUERY",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "me",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 60,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
